src/charm.py

```diff
@@ -184,16 +184,23 @@ def _reconcile_service(self) -> None:
         service = self._get_service()
         service_exists = service is not None
         service_type = service_exists and _ServiceType(service.spec.type)
-        if service_exists and service_type == desired_service_type:
-            return
-
-        pod0 = self._get_pod(f"{self.app.name}/0")
-
+        if service_exists:
+            existing_annotations = service.metadata.annotations or {}
+        else:
+            existing_annotations = {}
```

Comment:
If we are going to do `... or {}` even if the service exists, should we skip the conditional altogether?

```suggestion
        existing_annotations = service.metadata.annotations or {}
```

Reply:
If the service doesn't exist, it will give an error, because it tries to access the `.metadata.annotations`
This was proven by the integration tests:
```
  File "/var/lib/juju/agents/unit-mysql-router-k8s-0/charm/src/charm.py", line 187, in _reconcile_service
    existing_annots = service.metadata.annotations or {}
AttributeError: 'NoneType' object has no attribute 'metadata'
```

Reply:
Got it. Thanks! 👍🏻 

Style nit: what if we give the service annotations an empty dictionary as default value, and populate it when they actually exists? We can also follow the naming convention in the lines above (i.e. use `service_` prefix).

```suggestion
        service_annotations = {}
        if service_exists:
            service_annotations = service.metadata.annotations or {}
```

Reply:
Totally agree! Updated the commit now